comms/storage/longterm/longterm.go

```diff
@@ -0,0 +1,158 @@
+		ReplayPolicy:  nats.ReplayInstantPolicy,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Create a subscription on the consumer for every node
+	// Subject can be empty since it defaults to all subjects bound to the stream
+	storerSub, err := lt.jsc.PullSubscribe("", storerDurable, nats.BindStream(uploadStream))
+	if err != nil {
+		panic(err)
+	}
+
+	// Watch KV store to download files to long-term storage
+	// TODO: Maybe there should be an exit channel for in case we restart StorageServer without restarting the whole program? (e.g., if we want to update StorageDecider to pass it a new slice of storage node pubkeys)
+	go func() {
+		for {
+			msgs, err := storerSub.Fetch(1)
+			if err == nil {
+				msgs[0].Ack()
+
+				job := transcode.Job{}
+				err := json.Unmarshal(msgs[0].Data, &job)
+				if err != nil {
+					panic(err)
+				}
+
+				if job.Status == transcode.JobStatusDone {
+					if lt.storageDecider.ShouldStore(job.ID) {
+						fmt.Printf("Storing file with ID %q\n", job.ID)
+						lt.moveTempToLongTerm(job.Results)
+					} else {
+						fmt.Printf("Not storing file with ID %q\n", job.ID)
+						continue
+					}
+				}
+			} else if err != nats.ErrTimeout { // Timeout is expected when there's nothing new in the stream
+				fmt.Printf("Error fetching message to store a file: %q\n", err)
+			}
+		}
+	}()
+}
+
+func (lt *LongTerm) moveTempToLongTerm(tmpObjects []*nats.ObjectInfo) {
+	// Open the long-term storage *blob.Bucket
+	ctx := context.Background()
+	bucket, err := blob.OpenBucket(ctx, getBucketUrl())
+	if err != nil {
+		fmt.Printf("error: failed to store file - could not open bucket: %v", err)
+		return
+	}
+	defer bucket.Close()
+
+	for _, tmpObj := range tmpObjects {
+		// Get object from temp store
+		objStore, err := lt.jsc.ObjectStore(tmpObj.Bucket)
+		if err != nil {
+			log.Printf("Failed to get object from temp store: %v\n", err)
+			return
+		}
+		tempObj, err := objStore.Get(tmpObj.Name)
+		if err != nil {
+			log.Printf("Failed to get object from temp store: %v\n", err)
+			return
+		}
+
+		// Open a *blob.Writer for the blob at key=tmpObj.Name
+		writer, err := bucket.NewWriter(ctx, tmpObj.Name, nil)
+		if err != nil {
+			log.Printf("Failed to write %q: %v\n", tmpObj.Name, err)
+			return
+		}
+		defer writer.Close()
+
+		// Copy the data
+		_, err = io.Copy(writer, tempObj)
+		if err != nil {
+			log.Printf("Failed to copy data: %v\n", err)
+			return
+		}
+	}
+}
+
+// getBucketUrl returns the URL for the long-term storage bucket - default file storage at /tmp/audius-long-term.
+func getBucketUrl() string {
+	return "file://" + getFileBucketPath()
+}
+
+func getFileBucketPath() string {
+	rootDir := "/tmp/audius-long-term"
+	// On Windows, convert "\" to "/" (for in case rootDir is an env var in the future) and add a leading "/":
+	slashDir := filepath.ToSlash(rootDir)
+	if os.PathSeparator != '/' && !strings.HasPrefix(slashDir, "/") {
+		slashDir = "/" + slashDir
+	}
+	return slashDir
```

Comment:
could be like:

```
newpath := filepath.Join(os.TempDir(), "audius-long-term")
err := os.MkdirAll(newpath, os.ModePerm)
```

Reply:
nice, that's much cleaner. I made this change but now thinking it should be out of the temp dir because some SPs might actually use file storage. but maybe that's okay if it's all run in docker and mapped to a volume anyway?

Reply:
true... yeah it should be an ENV variable and our code can fall back to `os.TempDir()` if it's not set or something like that.